### ISSUE TITLE
"Description" in Hosts List always empty (still)

### ISSUE THREAD

**Issue by karpiyon** [Ettercap/ettercap#1164]
This has been discussed and and I tried to run with -d option but i do not see any description.
I run it on Kali 2021 4, running on VB on win10 host.
I noticed that the first time I show the scan results, on the description tab, is show "Resolving..." for some IPs but blanks a second later.
My description for all hosts is empty.

When I run bettercap... net.show I do see the description.

**Comment by koeppea** [comment]
I tried to reproduce with also on Kali 2021 4 on Virtual Box.
Reverse name resolution works and the Description column is populated with data as expected.
Can you purge ettercap and compile ettercap from Github source in Debug mode?

**Comment by karpiyon** [comment]
Yes, I'll do that, probably tomorow.
You prefer I purge & compile, over purge & apt-get install since the version in the repository may not be the most recent one?

**Comment by karpiyon** [comment]
trying to install, following teh INSTALL instructions i get:
```
CMake Error at cmake/Modules/FindGTK3.cmake:382 (message):
  Could not find GTK3 include directory
Call Stack (most recent call first):
  cmake/Modules/EttercapLibCheck.cmake:53 (find_package)
  CMakeLists.txt:106 (include)
```
with these errors in teh log:
```

Determining if the include file ltdl.h exists failed with the following output:
Change Dir: /home/gali/Downloads/ettercap/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_1afd1/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_1afd1.dir/build.make CMakeFiles/cmTC_1afd1.dir/build
gmake[1]: Entering directory '/home/gali/Downloads/ettercap/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_1afd1.dir/CheckIncludeFile.c.o
/usr/bin/cc    -o CMakeFiles/cmTC_1afd1.dir/CheckIncludeFile.c.o -c /home/gali/Downloads/ettercap/build/CMakeFiles/CMakeTmp/CheckIncludeFile.c
/home/gali/Downloads/ettercap/build/CMakeFiles/CMakeTmp/CheckIncludeFile.c:1:10: fatal error: ltdl.h: No such file or directory
    1 | #include <ltdl.h>
      |          ^~~~~~~~
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTC_1afd1.dir/build.make:78: CMakeFiles/cmTC_1afd1.dir/CheckIncludeFile.c.o] Error 1
gmake[1]: Leaving directory '/home/gali/Downloads/ettercap/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:127: cmTC_1afd1/fast] Error 2
```

**Comment by karpiyon** [comment]
what i did is:
```
git clone https://github.com/Ettercap/ettercap.git
cd ettercap
mkdir build
cd build
cmake ../
make
```
this is were got stuck

**Comment by koeppea** [comment]
You're lacking some dependencies. Check out the README. It contains an apt-get call to install the required dependencies for compiling ettercap.

**Comment by karpiyon** [comment]
I installed all missing dependencies and compiled ettercap : **ettercap 0.8.4-rc** 
I used: cmake -DCMAKE_BUILD_TYPE=Debug ..
I run it again and the description section is still empty.

**Comment by koeppea** [comment]
In the directory where ettercap has been executed, a ettercap debug log file should be there.
Can you please upload this file?

**Comment by karpiyon** [comment]
Sorry but i can't find the log.
You say it should be in the directory from which i run ettercap?
I could not find any log file, neither in ettercap compile folders

**Comment by koeppea** [comment]
Yes. It's created in the directory where ettercap has been launched when ettercap has been compliled in Debug mode, which you wrote you did.
It's called `ettercap-<version>_debug.log`. So it should be `ettercap-0.8.4-rc_debug.log`. Should be able to be found with find: `find / -name 'ettercap-*_debug.log 2>/dev/null'

**Comment by karpiyon** [comment]
i can't find it...
perhaps it was not installed in debug?
can you send me a link to the doc which explains how to install in debug mode?
what i did was:
`cmake -DCMAKE_BUILD_TYPE=Debug ..`

**Comment by LocutusOfBorg** [comment]
https://github.com/Ettercap/ettercap/blob/master/README.BUGS

**Comment by karpiyon** [comment]
Recompiled it again with teh same options as before for debugging.
Still no log that i can find.
Tried find/locate...
Where should it be generated?

**Comment by koeppea** [comment]
Here is our [Wiki article](https://github.com/Ettercap/ettercap/wiki/Providing-debug-information) which I've dedicated to this topic.
That the debug logfile cannot be written can only be if you run ettercap as a user that doesn't have access to the current working directory. for example, if you issue `sudo ettercap [OPTIONS]` in your home directory, which is mounted as a NFS export with _root_squash_ attribute.

Can you please past here the output when you launch ettercap, beginning with the command that you execute.

**Comment by karpiyon** [comment]
Run as admin and here is the log:

[ettercap-0.8.4-rc_debug.zip](https://github.com/Ettercap/ettercap/files/7981709/ettercap-0.8.4-rc_debug.zip)

**Comment by koeppea** [comment]
I had a look a debug log.
By default, Ettercap does not do active name resolution by default as it tries not to become too obvious.
It only does passive name resolution (e.g. when a Multicast DNS packet flies by or if a DNS response from a victim is is forwarded).
But from the subject, I don#t think this is what you're looking for.

As per the log file you provided, you have not provided the `-d`  flag to enable active name resolution on the CLI.
You have also not activated active name resolution while ettercap is running:
![grafik](https://user-images.githubusercontent.com/4201853/152222596-9206b64c-d510-43e6-9660-9a229b9c46de.png)

So based on what I can read from the log file, it's clear why description column of the hosts list remains quite empty.
